keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,291 @@
+requests/limits of a Job into the resource requests of the Job's Workload.
+This direct mapping results in two significant limitations that limit the expressivity
+of Kueue's resource and quota mechanisms:
+1. For a Job to be admissible by a ClusterQueue, the ClusterQueue must have
+a ResourceGroup whose CoveredResources include all resources requested by the Job.
+2. Only resources that are explicitly listed in a Jobs requests/limits can
+be used in Kueue's quota calculations.
+
+Injecting a configurable transformation function into this mapping
+will enable Kueue to support simpler and more powerful definitions of resource quotas.
+
+### Goals
+
+* Establish a mechanism that enables limited transformations on resources
+  to be specified via external configuration.
+
+### Non-Goals
+
+* Supporting complex numerical transformations of resource quantities
+* Supporting transformations that take multiple input resources
+* Performing any mutation of the resource requests/limits of Jobs
+* Ensuring that a Workload's resource requests is updated if the set of 
+configured transformations functions changes after its resource requests are computed.
+
+## Proposal
+
+* Add a configuration mechanism that enables a cluster admin to specify a 
+`ResourceTransformation` matrix.
+* Extend the `workload.AdjustResources` function so that it applies this
+matrix when adjusting the resource request for its input Workload object.
+
+### User Stories (Optional)
+
+In the user stories below, we use a 4-tuple (InputResourceName, OutputResourceName, OutputQuantity, Replace)
+to express a resource transformation operation. In the actual implementation, this transformation would
+be encoded as an entry in the  `ResourceTransformation` matrix which would be created by
+a cluster admin.
+
+#### Story 1
+
+This KEP will improve Kueue's ability to manage families of closely related 
+extended resources without creating overly complex ResourceGroups in its 
+ClusterQueues.  A motivating example is managing quotas for the various 
+MIG resources created by the NVIDIA GPU Operator using the mixed strategy
+(see [MIG Mixed Strategy][mig]).
+In the mixed strategy, instead of there being a single `nvidia.com/gpu` extended resource, 
+there is a proliferation of extended resource of the form `nvidia.com/mig-1g.5gb`, `nvidia.com/mig-2g.10gb`,
+`nvidia.com/mig-4g.20gb`, and so on.  For quota management purposes, we would like to abstract
+these into a single extended resource `kueue.x-k8s.io/accelerator-memory` and completely elide
+the MIG extended resource from the quota calculation.
+
+Concretely, the cluster admin would register the following transformations:
+* (`nvidia.com/mig-1g.5gb`, `kueue.x-k8s.io/accelerator-memory`, `5G`, `true`)
+* (`nvidia.com/mig-2g.10gb`, `kueue.x-k8s.io/accelerator-memory`, `10G`, `true`)
+* (`nvidia.com/mig-4g.20gb`, `kueue.x-k8s.io/accelerator-memory`, `20G`, `true`)
+
+A Job requesting the resources below:
```

Comment:
I expanded the MIG user story into two pieces and tried to motivate why replace might be desirable as an option (mainly to DRY up the ClusterQueue's  ResourceGroup).  I don't think the replace/retain configuration is absolutely essential, but it could be useful in some cases.

Reply:
Looks great.